lib/src/undo.rs

```diff
@@ -59,43 +60,28 @@ impl YrsUndoManager {
         m.reset();
     }
 
-    pub(crate) fn observe_added(&self, delegate: Box<dyn YrsUndoManagerObservationDelegate>) -> u32 {
+    pub(crate) fn observe_added(&self, delegate: Box<dyn YrsUndoManagerObservationDelegate>) -> Arc<YSubscription> {
         let m = self.acquire_lock();
         let subscription = m.observe_item_added(move |_, e| {
-            e.item.meta = delegate.call(YrsUndoEvent::new(e), e.item.meta);
+            *e.meta_mut() = delegate.call(YrsUndoEvent::new(e), *e.meta_mut());
```

Comment:
Is this ok? I don't really understand what's going on here, I just wanted it to compile and this looked reasonable but I have no evidence that this is working as expected (other than tests passing).

Reply:
Yes, it's fine. `e.meta_mut()` returns a mutable reference to meta object of event (which here IIRC is a memory address to an object). `*e.meta_mut() = ...` is overriding this meta reference with new value (basically reassignment).

This way you can attach meta objects to undo events and pop them later on ie. remember cursor position when `undo` is being performed and then restore that cursor position from meta object during `redo` action.